packages/connect-thegraph/src/core/GraphQLWrapper.ts

```diff
@@ -76,31 +79,52 @@ export default class GraphQLWrapper {
         if (this.#verbose) {
           console.log(this.describeQueryResult(result))
         }
-
         if (result.error) {
-          throw new Error(
-            [
-              'Error performing subscription.',
-              `${result.error.name}: ${result.error.message}`,
-              this.describeQueryResult(result),
-            ].join('\n')
+          callback(
+            new Error(
+              'Error performing subscription.\n' +
+                `${result.error.name}: ${result.error.message} \n` +
+                this.describeQueryResult(result)
+            )
           )
+          return
```

Comment:
Would you mind explain why do we need to return here?

Reply:
It is so that it stops here in case of error, and doesn’t continue to call `callback(null, result)` which is right after.